tracing/tracing-test.sh

```diff
@@ -63,112 +103,132 @@ stop_jaeger()
-# agent.
-check_missing_spans()
+# Returns status from Jaeger web UI
+get_jaeger_status()
 {
 	local service="$1"
-	local min_spans="$2"
+	local logdir="$2"
 
 	[ -z "$service" ] && die "need jaeger service name"
-	[ -z "$min_spans" ] && die "need minimum trace span count"
+	[ -z "$logdir" ] && die "need logdir"
 
-	local logfile=$(mktemp)
-
-	sudo journalctl -q -o cat -a -t "$service" > "$logfile"
-	sudo chown "$USER:" "$logfile"
+	local attempt=0
+	local max_attempts=10
 
-	# This message needs to be logged by each Kata component, generally when
-	# debug is enabled.
-	local component_prefix="created span"
+	local status=""
+	local span_count=0
 
-	# Message prefix added by Jaeger when LogSpans=true
-	# (see
-	# https://godoc.org/github.com/uber/jaeger-client-go/config#ReporterConfig).
-	local jaeger_reporter_prefix="Reporting span"
+	# Loop until we find some spans
+	while [ "$attempt" -lt "$max_attempts" ]
+	do
+		status=$(get_jaeger_traces "$service" || true)
+		if [ -n "$status" ]; then
+			echo "$status" > "$logdir/${service}-status.json"
+			span_count=$(get_span_count "$status")
+			[ "$span_count" -gt 0 ] && break
+		fi
 
-	local logged_spans
-	if ! logged_spans=$(grep -E -o "${component_prefix} [^ ][^ ]*" "$logfile" | awk '{print $3}')
-	then
-		info "failed to check logged spans"
-		rm -f "$logfile"
-		return
-	fi
+		attempt=$((attempt+1))
+		sleep 1
+	done
 
-	if [ -z "$logged_spans" ]
-	then
-		info "No logged spans to check"
-		rm -f "$logfile"
-		return
-	fi
+	[ -z "$status" ] && die "no status"
+	[ "$span_count" -eq 0 ] && die "no spans"
 
-	local count=0
+	attempt=0
 
-	for span in $logged_spans
+	# How that we have a valid status, poll until it settles
+	while true
 	do
-		count=$((count+1))
+		# Yes, it's horrid but we need to pause to give the Jaeger
+		# web service time to think ;)
+		sleep 1
 
-		# Remove quotes
-		span=$(echo $span|tr -d '"')
+		local new_status=$(get_jaeger_traces "$service" || true)
 
-		grep -E -q "${jaeger_reporter_prefix} \<$span\>" "$logfile" || \
-			die "span $count ($span) not reported"
-	done
+		# We have already queried the services status, so it should
+		# still be there!
+		[ -z "$new_status" ] && die "status disappeared"
+
+		local new_span_count=$(get_span_count "$new_status")
+		[ -z "$new_span_count" ] && die "no span count"
+		[ "$new_span_count" -le 0 ] && die "invalid span count"
+		[ "$new_span_count" -lt "$span_count" ] && die "span count dropped ($span_count to $new_span_count)"
+
+		# Span count didn't change, so the service
+		# has "stabilised"
+		if [ "$new_span_count" -eq "$span_count" ]; then
+			get_trace_summary "$status" \
+				> "$logdir/span-summary.txt"
+			break
+		fi
 
-	[ "$count" -lt "$min_spans" ] && \
-		die "expected >= $min_spans reported spans, got $count"
+		# The span count increased, so reset
+		status="$new_status"
+		span_count="$new_span_count"
 
-	info "All $count spans reported"
+		[ "$attempt" -eq "$max_attempts" ] && \
```

Comment:
I failed to find where this `$attempt` is incremented.

Reply:
Fixed.